api4/config_test.go

```diff
@@ -146,7 +145,7 @@ func TestUpdateConfig(t *testing.T) {
 		require.Equal(t, SiteName, cfg.TeamSettings.SiteName, "It should update the SiteName")
 
 		t.Run("Should set defaults for missing fields", func(t *testing.T) {
-			_, appErr := th.SystemAdminClient.DoApiPut(th.SystemAdminClient.GetConfigRoute(), fmt.Sprintf(`{"ServiceSettings":{"SiteURL":"%s"}}`, *cfg.ServiceSettings.SiteURL))
+			_, appErr := th.SystemAdminClient.DoApiPut(th.SystemAdminClient.GetConfigRoute(), `{"ServiceSettings":{"SiteURL":""}}`)
```

Comment:
Wouldn't this cause a failure if the `SiteURL` is set? 
https://github.com/mattermost/mattermost-server/blob/f4ccc7061ca01ad41f3789d7d01443b2161f1c66/api4/config.go#L111-L114

Reply:
Yeah you're right. This should be other-way round. It turned out that my env var was being set after the config initializes. Therefore it passed my test suite 😓